x/staking/keeper/liquid_stake.go

```diff
@@ -441,3 +442,34 @@ func (k Keeper) RefreshTotalLiquidStaked(ctx sdk.Context) error {
 
 	return nil
 }
+
+// CheckVestedDelegationInVestingAccount verifies whether the provided vesting account
+// holds a vested delegation for an equal or greater amount of the specified coin
+// at the given block time.
+//
+// Note that this function facilitates a specific use-case in the LSM module for tokenizing vested delegations.
+// For more details, see https://github.com/cosmos/gaia/issues/2877.
+func CheckVestedDelegationInVestingAccount(
+	account vesting.VestingAccount,
+	blockTime time.Time,
+	coin sdk.Coin,
+) bool {
+	// Get the vesting coins at the current block time
+	vestingAmount := account.GetVestingCoins(blockTime).AmountOf(coin.Denom)
+
+	// Note that the "DelegatedVesting" and "DelegatedFree" values
+	// were computed during the last delegation or undelegation operation
+	delVestingAmount := account.GetDelegatedVesting().AmountOf(coin.Denom)
+	delVested := account.GetDelegatedFree()
+
+	// Calculate the new vested delegated coins
+	x := sdk.MinInt(vestingAmount.Sub(delVestingAmount), math.ZeroInt())
+
+	// Add the newly vested delegated coins to the existing delegated vested amount
+	if !x.IsZero() {
+		delVested = delVested.Add(sdk.Coin{Denom: coin.Denom, Amount: x.Abs()})
```

Comment:
nit: use `sdk.NewCoin`

Reply:
resolved in [cd5b8061](https://github.com/cosmos/cosmos-sdk/pull/19614/commits/cd5b806132b1a6271c43c4d92efed0ac6af4d454).